frontend/src/components/utils/tags/TagEdition.tsx

```diff
@@ -0,0 +1,300 @@
+            );
+
+            if (selectedNamespaces.length >= 1) {
+                // If there is a namespace in the selected option, we want to display the tags
+                let selectedNamespace = selectedNamespaces[0].namespace;
+                setSelectedNamespace(selectedNamespace);
+                setInputValue(`${selectedNamespace.name}: `);
+                setMenuIsOpen(true);
+            } else if (newTag.length >= 1) {
+                // If there is a namespace in the selected option, we want to display the tags
+                api.tags
+                    .create(newTag[0].value, selectedNamespace)
+                    .then((createdTag) => {
+                        bindTag(createdTag);
+                    })
+                    .catch(() => {
+                        newToast({
+                            message: "Erreur lors de l'ajout du tag.",
+                            level: ToastLevel.Error,
+                        });
+                    });
+
+                setSelectedNamespace(null);
+            } else {
+                // all tags in the selection options, we bind the lasted option added
+                bindTag(selectedOption[selectedOption.length - 1].tag);
+                setSelectedNamespace(null);
+            }
+        } else if (action.action === "remove-value") {
+            removeTag(action.removedValue.tag);
+        }
+    };
+
+    const onInputChange = (value, action) => {
+        setInputValue(value);
+
+        // Check if we still have a namespace selected
+        let searchValue = value;
+        if (value.indexOf(":") === -1) {
+            if (selectedNamespace !== null) {
+                setSelectedNamespace(null);
+            }
+        } else {
+            let namespaceValue = value.split(":")[0];
+            let matchingNamespaces = namespaces.filter(
+                (namespace) => namespace.name == namespaceValue
+            );
+            if (matchingNamespaces.length !== 0) {
+                if (selectedNamespace == null) {
+                    setSelectedNamespace(matchingNamespaces[0]);
+                }
+                searchValue = value.split(":")[1];
+            }
+        }
+
+        searchValue = searchValue.trim();
+        setSearchValue(searchValue);
+
+        // Refresh the different suggestions
+        if (selectedNamespace === null) {
+            searchValue.length === 0
+                ? namespaces.slice(0, 10)
+                : setSuggestionsNamespace(
+                      fuseNamespace!
+                          .search(searchValue, { limit: 10 })
+                          .map((suggestion) => suggestion.item)
+                  );
+        } else {
+            setSuggestionsTag(
+                fuseTag!
+                    .search(searchValue, { limit: 10 })
+                    .map((suggestion) => suggestion.item)
+            );
+        }
+    };
+
+    const filterOption = (candidate, input) => {
```

Comment:
```suggestion
    const filterOption = (candidate, input) => candidate.data.type !== "new_value" || searchValue.length > 0;
```

Reply:
It would be more like `(candidate.data.type === "new_value" && searchValue.length > 0) || candidate.data.type !== "new_value"` but it's less clear